src/runtime/program/precompile_programs/ed25519.zig

```diff
@@ -0,0 +1,297 @@
+        .pubkey_instruction_idx = 1,
+    };
+    try std.testing.expectEqual(
+        testCase(1, invalid_pubkey_offsets),
+        error.InvalidDataOffsets,
+    );
+}
+
+// https://github.com/anza-xyz/agave/blob/a8aef04122068ec36a7af0721e36ee58efa0bef2/sdk/src/ed25519_instruction.rs#L326
+test "ed25519 message data offsets" {
+    {
+        const offsets: Ed25519SignatureOffsets = .{
+            .message_data_offset = 99,
+            .message_data_size = 1,
+        };
+        try std.testing.expectEqual(
+            testCase(1, offsets),
+            error.InvalidSignature,
+        );
+    }
+
+    {
+        const offsets: Ed25519SignatureOffsets = .{
+            .message_data_offset = 100,
+            .message_data_size = 1,
+        };
+        try std.testing.expectEqual(
+            testCase(1, offsets),
+            error.InvalidDataOffsets,
+        );
+    }
+
+    {
+        const offsets: Ed25519SignatureOffsets = .{
+            .message_data_offset = 100,
+            .message_data_size = 1000,
+        };
+        try std.testing.expectEqual(
+            testCase(1, offsets),
+            error.InvalidDataOffsets,
+        );
+    }
+
+    {
+        const offsets: Ed25519SignatureOffsets = .{
+            .message_data_offset = std.math.maxInt(u16),
+            .message_data_size = std.math.maxInt(u16),
+        };
+        try std.testing.expectEqual(
+            testCase(1, offsets),
+            error.InvalidDataOffsets,
+        );
+    }
+}
+
+// https://github.com/anza-xyz/agave/blob/a8aef04122068ec36a7af0721e36ee58efa0bef2/sdk/src/ed25519_instruction.rs#L369
+test "ed25519 pubkey offset" {
+    {
+        const offsets: Ed25519SignatureOffsets = .{
+            .pubkey_offset = std.math.maxInt(u16),
+        };
+        try std.testing.expectEqual(
+            testCase(1, offsets),
+            error.InvalidDataOffsets,
+        );
+    }
+
+    {
+        const offsets: Ed25519SignatureOffsets = .{
+            .pubkey_offset = 100 - ED25519_PUBKEY_SERIALIZED_SIZE + 1,
+        };
+        try std.testing.expectEqual(
+            testCase(1, offsets),
+            error.InvalidDataOffsets,
+        );
+    }
+}
+
+// https://github.com/anza-xyz/agave/blob/a8aef04122068ec36a7af0721e36ee58efa0bef2/sdk/src/ed25519_instruction.rs#L389-L390
+test "ed25519 signature offset" {
+    {
+        const offsets: Ed25519SignatureOffsets = .{
+            .signature_offset = std.math.maxInt(u16),
+        };
+        try std.testing.expectEqual(
+            testCase(1, offsets),
+            error.InvalidDataOffsets,
+        );
+    }
+
+    {
+        const offsets: Ed25519SignatureOffsets = .{
+            .signature_offset = 100 - ED25519_SIGNATURE_SERIALIZED_SIZE + 1,
+        };
+        try std.testing.expectEqual(
+            testCase(1, offsets),
+            error.InvalidDataOffsets,
+        );
+    }
+}
+
+// TODO: should we implement https://github.com/anza-xyz/agave/pull/1876/ ?
```

Comment:
feels like we should have it - wdyt

Reply:
Left it in as a TODO, we should definitely support it eventually. 
Changed the comment in 41dbdff2327181564202bac43005ba2cf3959cfd.

Going to add that at a later date